src/rust/engine/src/intrinsics.rs

```diff
@@ -302,21 +302,19 @@ fn remove_prefix_request_to_digest(
   context: Context,
   args: Vec<Value>,
 ) -> BoxFuture<'static, NodeResult<Value>> {
-  let core = context.core;
-  let store = core.store();
   async move {
-    let (input_digest, raw_prefix) = Python::with_gil(|py| {
-      let py_remove_prefix = (*args[0]).as_ref(py);
-      let py_digest = externs::getattr(py_remove_prefix, "digest").unwrap();
-      let input_digest = lift_directory_digest(py_digest).map_err(throw)?;
-      let prefix: String = externs::getattr(py_remove_prefix, "prefix").unwrap();
-      let res: NodeResult<(Digest, String)> = Ok((input_digest, prefix));
-      res
+    let py_remove_prefix = Python::with_gil(|py| {
+      (*args[0])
+        .as_ref(py)
+        .extract::<PyRemovePrefix>()
```

Comment:
I'm intentionally not using `externs::getattr` in favor of using PyO3 directly. Less indirection.